games/devtest/mods/testtools/init.lua

```diff
@@ -634,6 +613,56 @@ minetest.register_tool("testtools:object_attacher", {
 	end,
 })
 
+minetest.register_tool("testtools:children_getter", {
+	description = S("Children Getter") .."\n"..
+		S("Shows list of objects attached to object") .."\n"..
+		S("Punch object to show its 'children'") .."\n"..
+		S("Punch air to show your own 'children'"),
+	inventory_image = "testtools_children_getter.png",
+	groups = { testtool = 1, disable_repair = 1 },
+	on_use = function(itemstack, user, pointed_thing)
+		if user and user:is_player() then
+			local name = user:get_player_name()
+			local selected_object
+			local self_name
+			if pointed_thing.type == "object" then
+				selected_object = pointed_thing.ref
+			elseif pointed_thing.type == "nothing" then
+				selected_object = user
+			else
+				return
+			end
+			if selected_object:is_player() then
+				self_name = "player '"..selected_object:get_player_name().."'"
+			elseif selected_object:get_luaentity() then
+				self_name = "LuaEntity '"..selected_object:get_entity_name().."'"
```

Comment:
`get_entity_name` is marked for deprecation for years. `selected_object:get_luaentity().name` seems to be the new/correct way.
Related discussion: https://irc.minetest.net/minetest-dev/2021-02-27#i_5790818

EDIT: Also this code is the same as in the loop below (except for the `"* "` prefix). A function would avoid this duplication, but the decision for this is up to you.